File: src/main/java/no/fintlabs/flyt/gateway/application/archive/dispatch/mapping/AdresseMappingService.java
package no.fintlabs.flyt.gateway.application.archive.dispatch.mapping;

import no.fint.model.resource.felles.kompleksedatatyper.AdresseResource;
import no.fintlabs.flyt.gateway.application.archive.dispatch.model.instance.AdresseDto;
import org.springframework.stereotype.Service;

import java.util.ArrayList;

@Service
public class AdresseMappingService {

    public AdresseResource toAdresseResource(AdresseDto adresseDto) {
        AdresseResource adresseResource = new AdresseResource();
        adresseDto.getAdresselinje().map(ArrayList::new).ifPresent(adresseResource::setAdresselinje);
        adresseDto.getPostnummer().ifPresent(adresseResource::setPostnummer);
        adresseDto.getPoststed().ifPresent(adresseResource::setPoststed);
        return adresseResource;
    }

}

File: src/main/kotlin/no/novari/flyt/archive/gateway/dispatch/mapping/AdresseMappingService.kt
package no.novari.flyt.archive.gateway.dispatch.mapping

import no.novari.fint.model.resource.felles.kompleksedatatyper.AdresseResource
import no.novari.flyt.archive.gateway.dispatch.model.instance.AdresseDto
import org.springframework.stereotype.Service

@Service
class AdresseMappingService {
    fun toAdresseResource(adresseDto: AdresseDto): AdresseResource =
        AdresseResource().apply {
            adresseDto.adresselinje?.let(::ArrayList)?.let(::setAdresselinje)
            adresseDto.postnummer?.let(::setPostnummer)
            adresseDto.poststed?.let(::setPoststed)
        }
}
